conch/analysis/autovot.py

```diff
@@ -0,0 +1,87 @@
+    if int(channels) != 1:
+        return False
+    return True
+
+def resample_for_autovot(soundfile, tmpdir):
+    output_file = os.path.join(tmpdir, "sound_file.wav")
+    subprocess.call(["sox", soundfile, "-c", "1", "-r", "16000", output_file])
+    return output_file
+    
+
+class MeasureVOTPretrained(object):
+    def __init__(self, classifier_to_use=None, min_vot_length=15, max_vot_length=250, window_max=30, window_min=30, debug=False):
+        if classifier_to_use is None:
+            raise ValueError("There must be a classifier to run AutoVOT")
+        else:
+            self.classifier_to_use = classifier_to_use
+        self.min_vot_length = min_vot_length
+        self.max_vot_length = max_vot_length
+        self.debug = debug
+        self.window_max = window_max
+        self.window_min = window_min
+
+    def __call__(self, segment):
+        file_path = os.path.expanduser(segment["file_path"])
+        begin = segment["begin"]
+        end = segment["end"]
+        vot_marks = sorted(segment["vot_marks"], key=lambda x: x[0])
+        grid = textgrid.TextGrid(maxTime=end)
+        vot_tier = textgrid.IntervalTier(name='vot', maxTime=end)
+        for vot_begin, vot_end, *extra_data in vot_marks:
+            vot_tier.add(vot_begin, vot_end, 'vot')
+        grid.append(vot_tier)
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            grid_path = "{}/file.TextGrid".format(tmpdirname)
+            csv_path = "{}/file.csv".format(tmpdirname)
+            wav_filenames = "{}/wavs.txt".format(tmpdirname)
+            textgrid_filenames = "{}/textgrids.txt".format(tmpdirname)
+
+            if not is_autovot_friendly_file(file_path):
+                file_path = resample_for_autovot(file_path, tmpdirname)
+
+            with open(wav_filenames, 'w') as f:
+                f.write("{}\n".format(file_path))
+
+            with open(textgrid_filenames, 'w') as f:
+                f.write("{}\n".format(grid_path))
+
+            grid.write(grid_path)
+            
+            if self.debug:
+                grid.write('/tmp/textgrid_from_conch.csv')
+                with open('/tmp/alt_wordlist.txt', 'w') as f:
+                    f.write("{}\n".format('/tmp/textgrid_from_conch.csv'))
+                subprocess.run(["auto_vot_decode.py", wav_filenames, '/tmp/alt_wordlist.txt', self.classifier_to_use, '--vot_tier', 'vot', '--vot_mark', 'vot', "--min_vot_length", str(self.min_vot_length), "--max_vot_length", str(self.max_vot_length), "--window_max", str(self.window_max), "--window_min", str(self.window_min)])
+            subprocess.run(["auto_vot_decode.py", wav_filenames, textgrid_filenames, self.classifier_to_use, '--vot_tier', 'vot', '--vot_mark', 'vot', '--csv_file', csv_path, "--min_vot_length", str(self.min_vot_length), "--max_vot_length", str(self.max_vot_length), "--window_max", str(self.window_max), "--window_min", str(self.window_min)])
```

Comment:
So this `auto_vot_decode.py` is just a script that's available on the path, right?  Does it still require python 2?  Regardless, we probably want to call python on it anyway (i.e., `subprocess.run(['python', 'auto_vot_decode.py', ...`) just to remove potential headaches where the script isn't executable.

Reply:
It does require python2. The one issue with calling it like this is that it requires the autovot/bin to be in the PATH for the script to run at all, if we call it like `subprocess.run(['python', 'auto_vot_decode.py', ...)` it would still need to have the autovot bin in the path, but now additionally the location of the script too.

Reply:
Right, yes, I think we could maybe go the route of the praat/reaper (i.e., you can specify a path to the auto_vot_decode.py or it assumes it's on the path).  I'm assuming the auto_vot_decode.py relies on relative paths from the auto_vot_script.py location for the binaries that it calls?

Reply:
Unfortunately not, it calls the binaries as though they're on the path, which is obviously not ideal(I could potentially change this in the autovot repo but not sure if I should).